store/treestore/tree.go

```diff
@@ -15,98 +15,124 @@
-	minlenKey  = len(hashPrefix) + 2 // at least sha512-aa
+	defaultPathPerm = os.FileMode(0770 | os.ModeSetgid)
+	defaultFilePerm = os.FileMode(0660)
+
+	hashPrefix = "sha256-"
 )
 
-// Store represents a store of rendered ACIs
+// Store represents a store of rendered images
 type Store struct {
-	dir string
-	// TODO(sgotti) make this an interface (acirenderer.ACIRegistry) when the ACIStore functions to update treestore size will be removed
-	store   *imagestore.Store
-	lockDir string
+	dir       string
+	renderDir string
+	// Path to previous implementaion render directory
+	// TODO(sgotti) remove when backward compatibility isn't needed anymore
+	oldRenderDir string
+	store        *imagestore.Store
+	db           *kvdb.DB
+	lockDir      string
 }
 
 func NewStore(dir string, store *imagestore.Store) (*Store, error) {
-	// TODO(sgotti) backward compatibility with the current tree store paths. Needs a migration path to better paths.
-	ts := &Store{dir: filepath.Join(dir, "tree"), store: store}
+	ts := &Store{dir: dir, renderDir: filepath.Join(dir, "tree"), store: store}
+
+	if err := os.MkdirAll(filepath.Dir(ts.dbpath()), defaultPathPerm); err != nil {
+		return nil, errwrap.Wrap(errors.New("cannot create treestore db dir"), err)
+	}
 
-	ts.lockDir = filepath.Join(dir, "treestorelocks")
+	ts.lockDir = filepath.Join(dir, "locks")
 	if err := os.MkdirAll(ts.lockDir, 0755); err != nil {
-		return nil, err
+		return nil, errwrap.Wrap(errors.New("cannot create treestore locks dir"), err)
+	}
+
+	if err := ts.initDB(); err != nil {
+		return nil, errwrap.Wrap(errors.New("cannot initialize treestore db"), err)
 	}
+	ts.db = kvdb.NewDB(ts.dbpath(), defaultFilePerm)
+
 	return ts, nil
 }
 
-// GetID calculates the treestore ID for the given image key.
-// The treeStoreID is computed as an hash of the flattened dependency tree
-// image keys. In this way the ID may change for the same key if the image's
-// dependencies change.
-func (ts *Store) GetID(key string) (string, error) {
-	hash, err := types.NewHash(key)
-	if err != nil {
-		return "", err
-	}
-	images, err := acirenderer.CreateDepListFromImageID(*hash, ts.store)
-	if err != nil {
-		return "", err
+func (ts *Store) dbpath() string {
+	return filepath.Join(ts.dir, "db", "db")
+}
+
+func (ts *Store) initDB() error {
+	dbpath := ts.dbpath()
+	db := kvdb.NewDB(dbpath, defaultFilePerm)
+
+	// Create the "info" bucket
+	if err := db.DoRW(func(tx *bolt.Tx) error {
+		_, err := tx.CreateBucketIfNotExists([]byte(infobucket))
+		return err
+	}); err != nil {
+		return err
 	}
+	return nil
+}
 
-	var keys []string
-	for _, image := range images {
-		keys = append(keys, image.Key)
+// SetOldTreeStoreDir sets the old treestore dir, used for backward compatibility
+// TODO(sgotti) remove when backward compatibility isn't needed anymore
+func (ts *Store) SetOldTreeStoreDir(olddir string) {
+	ts.oldRenderDir = olddir
+}
+
+// Returns all the treestore infos for the specified image digest
```

Comment:
I guess you'll rework all the docs but this doesn't start with the function name.